### ISSUE TITLE
Add a simple way in core to define private and public access for nodes/entities.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1407]
This idea sprang off of #1348 where two Drupal contrib modules were suggested as a possible solution to the issue reported. These modules are [Content Access](https://www.drupal.org/project/content_access) and [Simple Access](https://www.drupal.org/project/simple_access) ([Taxonomy Access Control](https://www.drupal.org/project/taxonomy_access) has also been mentioned). @biolithic has started a port of Content Access in https://github.com/biolithic/content_access and @Graham-72 ported Simple Access in https://github.com/backdrop-contrib/simple_access

It's worth mentioning as a reference https://github.com/backdrop-ops/contrib/issues/105 too.

In #1348 there was this idea also for what could be implemented in core:

From a brief look at the features of these two modules, I have to say that Simple Access seems more ...well "simple" and useful to me. But that might be because I have specific use-cases in mind that happen to match its features and its "mechanics" (I like the approach of leaving the rest of the content accessible by everyone instead of the other way around). [...]

From their UI screenshots available on d.org, I have to say that they seem a bit "too much" for core inclusion, but I still would like to see something like that offered by core. Perhaps a simplified version (UI-wise) could be put together. I was thinking something along the following lines perhaps:
1. A flag that simplifies the distinction of "private" (more fine-grained access control) vs "public" (_access content_ permission) nodes.
2. A per-content type setting that provides the following options:
   1. no privacy flag feature at all (the default for all content types).
   2. allow specific nodes of that type to be set as "private" (give authors an option).
   3. make it so that **all** nodes of that type are set to "private" (no option given to node author).
3. A set of permissions for allowing users to edit node privacy (allow edit all/own privacy flags).
4. A second set of permissions to allow specific role(s) to view all "private" nodes (in essence a permission to override this feature altogether - intended for admins).
5. A per-node "make this page/article private" checkbox (available to users with the respective permission from 3 above - provided to nodes of type that are set to option 2ii from above). 
6. Setting the private flag, would limit viewing only to the node author + users with the override permission from 4 above.
7. A bulk operation available to make nodes private en masse (credits for the idea to @Graham-72).
8. Contrib modules could extend this feature further. For example allow private nodes to be viewed by all users that have the same role(s) as the author (e.g allow some simplified version of group privacy) or OG integration (when ported).

Thoughts?

**Comment by Graham-72** [comment]
> @biolithic has started a port of Simple Access

A typo? - Actually this is a port of Content Access by @biolithic 

Perhaps you should also now reference my port of Simple Access https://github.com/backdrop-contrib/simple_access ?

**Comment by klonos** [comment]
I stand corrected. Typo indeed. Issue summary updated to correct to Content Access and also added and linked to your port of Simple Access.

**Comment by Graham-72** [comment]
Thanks @klonos for straightening the record. One day when I have some time I must do a comparison of Simple Access as it is now with your requirements list. Urgent need to get Mimemail working just now - not relevant to this issue!

**Comment by biolithic** [comment]
https://github.com/backdrop-contrib/mimemail is started -- it may be of use
to you

**Comment by Graham-72** [comment]
Yes, thanks @biolithic , I have forked your port of Mimemail as my starting point.

**Comment by klonos** [comment]
Adding this to #1388:

![wp-content_visibility](https://cloud.githubusercontent.com/assets/2423362/12071848/20d6848a-b116-11e5-93ff-03cadd179065.png)
![wp-content_visibility_options](https://cloud.githubusercontent.com/assets/2423362/12071849/25e9aeb6-b116-11e5-841f-e6de29aa5a0d.png)

**Comment by Graham-72** [comment]
@klonos when you have had some time to recuperate from the recent flurry of activity to get release 1.4 completed ( :+1:  :+1: ) I would be grateful if you could take a look at a new contrib module I have produced 'Simple Access Grants' and published at https://github.com/backdrop-contrib/simple_access_grants

Having looked at Content Access and ported Simple Access, I became convinced that something simpler was possible and would benefit from use of Backdrop's configuration system. The module I have produced does not give all the features you have described above, but most of them though not in quite the same way.

I look forward to any comments you may have. I shall be trying it in earnest on one of my sites that is currently using my port of Simple Access, making me aware of that module's shortcomings and a wish to do better.

**Comment by klonos** [comment]
Will definitely give it a go soon as I have some time @Graham-72 ...but it won't be soon (taking 9 days to reply to you should give you a hint, sorry).

**Comment by Graham-72** [comment]
@klonos no problem, I appreciate how busy you are. I have found and fixed some issues and now have it working on a client's site, but in view of your earlier interest it would be good if you can check it out when you have time.
